scapy/layers/dns.py

```diff
@@ -1046,6 +1051,55 @@ def pre_dissect(self, s):
 bind_layers(TCP, DNS, dport=53)
 bind_layers(TCP, DNS, sport=53)
 
+# Nameserver config
+conf.nameservers = read_nameservers()
+_dns_cache = conf.netcache.new_cache("dns_cache", 300)
+
+
+@conf.commands.register
+def dns_resolve(qname, qtype="A", verbose=1, **kwargs):
+    """
+    Perform a simple DNS resolution using conf.nameservers with caching
+    """
+    answer = _dns_cache.get("_".join([qname, qtype]))
+    if answer:
+        return answer
+
+    kwargs.setdefault("timeout", 5)
+    kwargs.setdefault("verbose", 0)
+    for nameserver in conf.nameservers:
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.connect((nameserver, 53))
+            sock = StreamSocket(sock, DNS)
+            res = sock.sr1(
+                DNS(qd=[DNSQR(qname=qname, qtype=qtype)]),
+                **kwargs,
+            )
+        except IOError as ex:
+            if verbose:
+                log_runtime.warning(str(ex))
+            continue
+        finally:
+            sock.close()
+        if res:
+            if res[DNS].rcode == 2:  # server failure
+                res = None
+                if verbose:
+                    log_runtime.info(
+                        "DNS: %s answered with failure for %s" % (
+                            nameserver,
+                            qname,
+                        )
+                    )
+            else:
+                break
+    if res is not None and res.an:
```

Comment:
you should also check that the qname and qtypes match the answer name and types. also, i don't know if it is safe to assume that the answer will always be in the first `an` element.

Reply:
Good call ! And you're right we should check them all.